Assets/Scripts/Gameplay/Mecha/MechaController.cs

```diff
@@ -334,8 +390,21 @@ private void OnMove(object data)
         {
             if (data is not Vector2 movement)
                 return;
+            CurrentMovementMode = MovementMode.Walking;
```

Comment:
You will face an issue later potentially.
If you dash and move right after, you will swap from Dashing to Walking in one frame,
which may not be what you expect

Reply:
I tried constantly breaking that when I was testing, nothing really happened, everything continued as usual, maybe theirs a better way to do it so theirs no potential risk, I'll look into it